src/Streamly/Internal/Prelude.hs

```diff
@@ -1993,10 +2029,9 @@ reverse' s = fromStreamD $ D.reverse' $ toStreamD s
 
 -- intersperseM = intersperseBySpan 1
 
--- | Generate a stream by performing a monadic action between consecutive
--- elements of the given stream.
---
--- /Concurrent (do not use with 'parallely' on infinite streams)/
+-- | Generate a stream by inserting the result of a monadic action between
+-- consecutive elements of the given stream. Note that the monadic action is
+-- performed after the stream action before which its result is inserted.
```

Comment:
This ceased to be concurrent when StreamD based implementation was done and we changed from K.intersperseM to S.intersperseM.

We can make it concurrent by using a rewrite rule but I think concurrency for this may not be very useful.